src/vcpkg/commands.cpp

```diff
@@ -1,189 +1,126 @@
+        {"z-applocal", z_applocal_command_and_exit},
 #endif // defined(_WIN32)
+        {"z-preregister-telemetry", z_preregister_telemetry_and_exit},
+    };
 
-        static std::vector<PackageNameAndFunction<const BasicCommand*>> t = {
-            {"version", &version},
-            {"contact", &contact},
-            {"hash", &hash},
-            {"x-init-registry", &init_registry},
-            {"x-download", &xdownload},
-            {"x-generate-default-message-map", &generate_message_map},
-            {"bootstrap-standalone", &boostrap_standalone},
-            {"z-preregister-telemetry", &zpreregister_telemetry},
-#if defined(_WIN32)
-            {"x-upload-metrics", &upload_metrics},
-            {"z-applocal", &applocal}
-#endif // defined(_WIN32)
-        };
-        return t;
-    }
+    constexpr Span<const PackageNameAndFunction<BasicCommandFn>> basic_commands = basic_commands_storage;
 
-    Span<const PackageNameAndFunction<const PathsCommand*>> get_available_paths_commands()
-    {
-        static const AcquireCommand acquire{};
-        static const AcquireProjectCommand acquire_project{};
-        static const ActivateCommand activate{};
-        static const AddCommand add{};
-        static const AddVersion::AddVersionCommand add_version{};
-        static const Autocomplete::AutocompleteCommand autocomplete{};
-        static const Cache::CacheCommand cache{};
-        static const CIClean::CICleanCommand ciclean{};
-        static const CIVerifyVersions::CIVerifyVersionsCommand ci_verify_versions{};
-        static const Create::CreateCommand create{};
-        static const DeactivateCommand deactivate{};
-        static const Edit::EditCommand edit{};
-        static const Fetch::FetchCommand fetch{};
-        static const FindCommand find_{};
-        static const FormatManifest::FormatManifestCommand format_manifest{};
-        static const GenerateMSBuildPropsCommand generate_msbuildprops{};
-        static const Help::HelpCommand help{};
-        static const Info::InfoCommand info{};
-        static const Integrate::IntegrateCommand integrate{};
-        static const List::ListCommand list{};
-        static const NewCommand new_{};
-        static const Owns::OwnsCommand owns{};
-        static const PortsDiff::PortsDiffCommand portsdiff{};
-        static const RegenerateCommand regenerate{};
-        static const SearchCommand search{};
-        static const Update::UpdateCommand update{};
-        static const UpdateBaselineCommand update_baseline{};
-        static const UseCommand use{};
-        static const X_VSInstances::VSInstancesCommand vsinstances{};
-        static const ZCeCommand ce{};
+    static constexpr PackageNameAndFunction<PathsCommandFn> paths_commands_storage[] = {
+        {"acquire", acquire_command_and_exit},
+        {"acquire-project", acquire_project_command_and_exit},
+        {"activate", activate_command_and_exit},
+        {"add", add_command_and_exit},
```

Comment:
I keep reading these as "Add a command and then exit"

Can we make the prefix common and have different suffixes, such as:
`perform_add_and_exit` or `command_add`

Reply:
I am still an fan of `Xxx::perform_and_exit` :)

Reply:
> I am still an fan of `Xxx::perform_and_exit` :)

My view on namespaces is 'if it looks like an error to call the function without namespace qualification, then the thing you put in the namespace should be in the name of the function'. I'm also not a huge fan of the word 'perform' here because it conveys nothing semantic; the word command here is to convey that it's the same as `vcpkg command`.